### PR TITLE
fix(datepicker): convert years 0-99 correctly with native adapters

### LINKED ISSUE #2731: Native Date adapter doesn't handle years < 100 correctly
See demo: https://ng-bootstrap.github.io/#/components/datepicker/examples#adapter

screen shot 2018-09-20 at 15 10 56

The year should be `10`, not `1910`.

### PR DESCRIPTION
**Pull request by maxokorokov**
Fixes #2731

### CODE REVIEW

File: src/datepicker/adapters/ngb-date-native-adapter.ts [hunk 1]
```diff
@@ -9,6 +10,16 @@ export class NgbDateNativeAdapter extends NgbDateAdapter<Date> {
   }
 
   toModel(date: NgbDateStruct): Date {
-    return date && date.year && date.month ? new Date(date.year, date.month - 1, date.day, 12) : null;
+    if (date && isNumber(date.year) && isNumber(date.month) && isNumber(date.day)) {
+      const jsDate = new Date(date.year, date.month - 1, date.day, 12);
+
+      // this is done avoid 30 -> 1930 conversion
+      if (!isNaN(jsDate.getTime())) {
```
Comment (pkozlowski-opensource) [inline src/datepicker/adapters/ngb-date-native-adapter.ts]: Couldn't we test for `date.year` < 100 (or whatever cut off point for conversion is)?

File: src/datepicker/adapters/ngb-date-native-utc-adapter.ts [hunk 2]
```diff
@@ -14,6 +15,16 @@ export class NgbDateNativeUTCAdapter extends NgbDateAdapter<Date> {
   }
 
   toModel(date: NgbDateStruct): Date {
-    return date && date.year && date.month ? new Date(Date.UTC(date.year, date.month - 1, date.day)) : null;
+    if (date && isNumber(date.year) && isNumber(date.month) && isNumber(date.day)) {
```
Comment (pkozlowski-opensource) [inline src/datepicker/adapters/ngb-date-native-utc-adapter.ts]: There is quite a bit of code duplication between adapters - could you think of a way to remove it?

File: src/datepicker/adapters/ngb-date-native-adapter.ts (hunk 1 above)
Comment (maxokorokov) [reply src/datepicker/adapters/ngb-date-native-adapter.ts]: Actually I was doing it to check that the `jsDate` is a valid date object. I guess more correct version would be:

```ts
if (jsDate instanceof Date && !isNan(jsDate) && date.year >= 0 && date.year <= 99) { ... }
```

But I kinda can omit `instanceof` part.

Do you want it to be like this ?
```ts
if (!isNan(jsDate) && date.year >= 0 && date.year <= 99) { ... }
```

File: src/datepicker/adapters/ngb-date-native-adapter.spec.ts [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import {NgbDateNativeAdapter} from './ngb-date-native-adapter';
+import {type} from 'os';
```
Comment (pkozlowski-opensource) [inline src/datepicker/adapters/ngb-date-native-adapter.spec.ts]: What is this import? I guess it was added by accident?

Comment (maxokorokov) [reply src/datepicker/adapters/ngb-date-native-adapter.spec.ts]: Yep, thanks

**Comment by pkozlowski-opensource** [comment]
LGTM!

**Pull request merged**